Tests/ChessGameTest.php

```diff
@@ -244,6 +244,34 @@ public function testCompleteGame()
         $this->assertEquals($endFen, $this->game->renderFen());
     }
 
+    /**
+     * Test two knights that can both move to the same square, but one is pinned to the king.
+     *
+     * 1.e4 e5 2.Bc4 Nf6 3.d3 Nc6 4.Nc3 Bb4 5.Ne2
+     */
+    public function testAmbiguousKnightMoves()
```

Comment:
The problem I see with this test is that it passes in latest develop. The test should fail in latest develop and pass with your changes in this branch.